phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/formats/hpo/HpoAssociationData.java

```diff
@@ -0,0 +1,48 @@
+package org.monarchinitiative.phenol.annotations.formats.hpo;
+
+import org.monarchinitiative.phenol.annotations.formats.GeneIdentifier;
+import org.monarchinitiative.phenol.ontology.data.TermId;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+// TODO - evaluate the utility of this container and investigate if it is possible to split the container to smaller parts.
+// TODO - should this be a one-stop-shop for all association data? If yes, then we should add HpoDiseases.
+public interface HpoAssociationData {
+
+  static HpoAssociationData of(List<GeneIdentifier> geneIdentifiers,
+                               Map<TermId, Collection<GeneIdentifier>> diseaseToGenes,
+                               Map<TermId, Collection<TermId>> geneToDiseases,
```

Comment:
My motivation for creating `GeneIdentifier` was that the previously used `Gene` was simply not correct. So now we have `GeneIdentifier` which consists of a `TermId id` and a `String hgvsSymbol`. I think it would be good to create `DiseaseIdentifier` and similar classes/interfaces for naming entities, instead of plain `TermId`, as the latter is prone to bugs. If that's the case, I think we should make something like `Labeled` or `Identified` interface, that would look like this:

```java
public interface Identified {
   TermId id();
}
```
In plain language, everything that *has* `TermId` should implement it, including `GeneIdentifier`, `DiseaseIdentifier`, but also `TermAnnotation`, `DiseaseToGeneAssociation`.. Phenol is full of labeled things, but finding what the primary term is is usually harder than it should be. I think that adding this can reduce the number of lines & make classes easier to understand.

@pnrobinson @julesjacobsen do you see any value in this approach? Should I go for it?

Reply:
This does seem a great idea!

Reply:
Sounds like a good plan to me too.

Reply:
Hi @pnrobinson  @julesjacobsen , I added the `Identified` interface and I implemented it in all classes that I considered relevant.